### PR TITLE
Migrate Basic Auth to bcrypt user login and cookie-backed sessions

### PR DESCRIPTION
**Pull request by dimroc**

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.2.0
+0.3.0
```
Comment (se3000) [inline VERSION]: 🍾

File: internal/fixtures/cookie [hunk 2]
```diff
@@ -0,0 +1 @@
+clsession=MTUzMTkyMzI0NHxEdi1CQkFFQ180SUFBUkFCRUFBQVJ2LUNBQUVHYzNSeWFXNW5EQTRBREdOc2MyVnpjMmx2Ymw5cFpBWnpkSEpwYm1jTUlnQWdNR016TWpobVptWmhaVFkxTkRNd05UaGpNelF3WXpZMFlqTTJNVGM1WkRBPXzDcp8k2N2ylrXA_3AQZlqiYpQqJIZOlpnh9XkQnj2HFQ==; Path=/; Expires=Fri, 17 Aug 2018 14:14:04 GMT; Max-Age=2592000
```
Comment (se3000) [inline internal/fixtures/cookie]: Do we have to worry about this expiring?

File: cmd/local_client.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	logger.Infow(balance)
+}
+
+func logConfigVariables(config strpkg.Config) {
+	wlc := presenters.NewConfigWhitelist(config)
+	logger.Debug("Environment variables\n", wlc)
+}
+
+// DeleteUser is run locally to remove the User row from the node's database.
+func (cli *Client) DeleteUser(c *clipkg.Context) error {
```
Comment (se3000) [inline cmd/local_client.go]: Seems worth adding an extra confirmation prompt, warning that all sessions will be killed if you continue.

Comment (dimroc) [reply cmd/local_client.go]: 👍 hopefully we can save it for another story/pr.

File: .gitignore [hunk 4]
```diff
@@ -3,12 +3,6 @@ node_modules/
 vendor/
 tmp/
 
-# chainlink tmp
```
[the remark this answers is not in the document]
Comment (dimroc) [reply .gitignore]: Moved it into `.gitignore`s closer to content.

File: internal/fixtures/cookie (hunk 2 above)
Comment (dimroc) [reply internal/fixtures/cookie]: No, it's merely a fixture for deserializing, not actual authentication.

**Pull request merged**